hastexo/tasks.py

```diff
@@ -165,14 +167,51 @@ def run(self, **kwargs):
 
         # Don't wait for the user to check results.  Update the database
         # immediately.
-        self.update_stack(stack_data)
+        self.update_stack_retry(stack_data)
 
     @transaction.atomic
     def update_stack(self, data):
         stack = Stack.objects.select_for_update().get(id=self.stack_id)
         update_stack_fields(stack, data)
         stack.save(update_fields=list(data.keys()))
 
+    def update_stack_retry(self, data):
+        """A wrapper around update_stack that retries until it either
+        succeeds, or times out."""
+
+        # This wraps a try/except block *around* update_stack, which
+        # uses transaction.atomic().
+        # Catching exceptions *inside* transaction.atomic is highly
+        # discouraged.
+        #
+        # Reference:
+        # https://docs.djangoproject.com/en/1.11/topics/db/transactions/
+        while True:
+            try:
+                # Try updating the stack. If it succeeds, we're done.
+                self.update_stack(data)
+                break
+            except SoftTimeLimitExceeded:
```

Comment:
Not sure if we even want to catch this here. We can't do cleanup, which is what we normally do when we hit `SoftTimeLimitExceeded`, so we may just want to throw this up the stack.

Reply:
It's still better to be able to try doing something, when all has gone up the creek.  I'd leave it in.

Reply:
Ack, thanks. Works for me.